Install pca-gophish-composition into a Python virtual environment

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the Ansible role to install pca-gophish-composition into a Python virtual environment.

## 💭 Motivation and context ##

Resolves #50.  See also cisagov/pca-gophish-composition-packer#105.  As part of the testing for cisagov/pca-gophish-composition-packer#105 I built a staging AMI with these changes, and I was able to confirm that the `pca-gophish-composition` service started up normally.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Install pca-gophish-composition into a Python virtual environment

This resolves #50.

dv4harr10 commented:
Hi Team, One small issue I notice for molecule/default/tests/test_default.py at line 11 I ran via VsCode and get the error message for Molecule_Inventory_File which suggests 'Add type annotations for all fixture references in this file'.

jsf9k commented:
> Hi Team, One small issue I notice for molecule/default/tests/test_default.py at line 11 I ran via VsCode and get the error message for Molecule_Inventory_File which suggests 'Add type annotations for all fixture references in this file'.

I'm not sure exactly what you are saying.  If we want to add type annotations to the test code then this should be done in [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).  @dv4harr10 - Please create an issue there, and better explain what change you are looking for.

dv4harr10 commented:
Hi Team, One small issue I notice for molecule/default/tests/test_default.py at line 11 I ran via VsCode and get the error message for Molecule_Inventory_File which suggests 'Add type annotations for all fixture references in this file'.

I'm not sure exactly what you are saying. If we want to add type annotations to the test code then this should be done in [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role). @dv4harr10 - Please create an issue there, and better explain what change you are looking for.

Hi Team, This is a KeyError I'm receiving from VsCode debugger for MOLECULE_INVENTORY_FILE. I'm not finding any supporting duplicate occurrences of this issue. Otherwise, this PR has no issues. Thanks

dav3r commented on molecule/default/tests/test_default.py:
```diff
@@ -42,4 +42,8 @@ def test_packages(host, pkg):
 def test_pip_packages(host, pkg):
     """Test that the pip packages were installed."""
-    assert pkg in host.pip.get_packages(pip_path="pip3")
+    # Note that we are using the version of pip3 in the Python virtual
```
Not sure if it makes any difference since pip is pip3 here, so feel free to ignore:
```suggestion
    # Note that we are using the version of pip in the Python virtual
```

dav3r reviewed (approved):
👍 👍 

FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.

jsf9k commented:
> 👍 👍
> 
> FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.

Is this a question that can be answered?  Who would know?

jsf9k committed:
Correct reference to pip3

`pip` and `pip3` are the same thing here, but we are explicitly calling `pip`.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

michaelsaki reviewed (approved):
Seems good to me!

dav3r commented:
> > 👍 👍
> > FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.
> 
> Is this a question that can be answered? Who would know?

I think you'd have to ask our friends in Assessments - perhaps @m1j09830 can find out or point us in the right direction.

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -29,7 +29,7 @@ def test_command(host, f):
 
 
-@pytest.mark.parametrize("pkg", ["at", "jq"])
+@pytest.mark.parametrize("pkg", ["at", "jq", "python3-virtualenv"])
```
If we're checking for `python3-virtualenv` then I think that is what we should be installing (instead of `virtualenv`).

jsf9k replied:
You are correct, sir!  Please let me know if commit d157ad8 is to your liking.

mcdonnnj replied:
Bonege!

m1j09830 commented:
> > > 👍 👍
> > > FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.
> > 
> > 
> > Is this a question that can be answered? Who would know?
> 
> I think you'd have to ask our friends in Assessments - perhaps @m1j09830 can find out or point us in the right direction.

Hey @dav3r, what tools are you referring to exactly. Sorry I tried reading through the comments to pin point what is being referenced but I'm lost on the ask here :(

jsf9k commented:
> > > > 👍 👍
> > > > FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.
> > > 
> > > 
> > > Is this a question that can be answered? Who would know?
> > 
> > 
> > I think you'd have to ask our friends in Assessments - perhaps @m1j09830 can find out or point us in the right direction.
> 
> Hey @dav3r, what tools are you referring to exactly. Sorry I tried reading through the comments to pin point what is being referenced but I'm lost on the ask here :(

We are specifically asking about `gophish-init` from [cisagov/pca-gophish-composition](https://github.com/cisagov/pca-gophish-composition/).  I'd also be interested to know if [the tools in the `gophish-tools` directory of the same repo](https://github.com/cisagov/pca-gophish-composition/tree/develop/gophish-tools) are being used.

dav3r commented:
> > > > > 👍 👍
> > > > > FWIW, I'm not sure if anyone is actively using those gophish-tools helper scripts anymore.
> > > > 
> > > > 
> > > > Is this a question that can be answered? Who would know?
> > > 
> > > 
> > > I think you'd have to ask our friends in Assessments - perhaps @m1j09830 can find out or point us in the right direction.
> > 
> > 
> > Hey @dav3r, what tools are you referring to exactly. Sorry I tried reading through the comments to pin point what is being referenced but I'm lost on the ask here :(
> 
> We are specifically asking about `gophish-init` from [cisagov/pca-gophish-composition](https://github.com/cisagov/pca-gophish-composition/). I'd also be interested to know if the tools in the `gophish-tools` directory of the same repo are being used.

@jsf9k One place that I think we need to update is [here](https://github.com/cisagov/pca-gophish-composition-packer/blob/6bf37a171a9ce021f54a2b5d390c26a6eec8d322/src/pca-gophish-composition.service#L16-L17).  I'm not sure if anyone else uses that `gophish-init` script.

jsf9k commented:
> @jsf9k One place that I think we need to update is [here](https://github.com/cisagov/pca-gophish-composition-packer/blob/6bf37a171a9ce021f54a2b5d390c26a6eec8d322/src/pca-gophish-composition.service#L16-L17). I'm not sure if anyone else uses that `gophish-init` script.

Ah, OK.  I will update that now.

jsf9k committed:
Install python3-virtualenv vice virtualenv

virtualenv is a virtual package that really just installs
python3-virtualenv anyway.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Treat Debian Buster as a special case

Debian Buster requires a system package that the other platforms do
not.

mcdonnnj reviewed (approved):
Sorry for the extra complication but LGTM ✔
